src/db.c

```diff
@@ -154,11 +135,17 @@ robj *lookupKeyRead(redisDb *db, robj *key) {
 /* Lookup a key for write operations, and as a side effect, if needed, expires
  * the key if its TTL is reached.
  *
+ * Flags change the behavior of this command:
+ *
+ *  LOOKUP_NONE (or zero): no special flags are passed.
+ *  LOOKUP_NOTOUCH: don't alter the last access time of the key.
+ *  LOOKUP_NOEXPIRE: don't check if the key has expired.
+ *
  * Returns the linked value object if the key exists or NULL if the key
  * does not exist in the specified DB. */
 robj *lookupKeyWriteWithFlags(redisDb *db, robj *key, int flags) {
-    expireIfNeeded(db,key);
-    return lookupKey(db,key,flags);
+    flags |= LOOKUP_WRITE | LOOKUP_NOSTATS | LOOKUP_NONOTIFY;
```

Comment:
i think that the `NOSTATS` and `NONOTIFY` where both innocent bugs.
people added code to lookupKeyReadWithFlags and forgot to update lookupKeyWrite..
i think we should drop these flags, and modify the top comment of the PR (to state the behavior change).

The stats are certainly not a breaking change (just stats).
the key-miss notification change should just be reflected in the release notes.

Reply:
ok, that's a good point, but:
first, in that case the source key of SUNIONSTORE we do want to count it as a miss, right?
I'd argue that for INCR we want that too.

you have a good point about SET, but then i'd say: great, so let's add separate stats, and even notification about write-misses.
if we do that, we can simplify things and consider INCR a write-miss.

Reply:
I've added TODO comments for write stats in lookupKey.